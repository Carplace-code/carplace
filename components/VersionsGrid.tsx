import VersionCard, { MinimumGridVersion } from "./VersionCard";

interface VersionsGridProps {
  versions: MinimumGridVersion[];
}

export default function VersionsGrid({ versions }: VersionsGridProps) {
  if (versions.length === 0) return <p className="py-6 text-center">No hay versiones disponibles</p>;

  return (
    <div className="grid w-full grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4">
      {versions.map((v) => (
        <VersionCard key={v.id} version={v} />
      ))}
    </div>
  );
}
