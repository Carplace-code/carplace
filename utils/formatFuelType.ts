import { FuelType } from "@prisma/client";

export default function formatFuelType(fuelType: FuelType): string {
  switch (fuelType) {
    case FuelType.gas:
      return "Gasolina";
    case FuelType.diesel:
      return "Diésel";
    case FuelType.electricity:
      return "Eléctrico";
    case FuelType.hybrid:
      return "Híbrido";
    default:
      return "Desconocido";
  }
}
