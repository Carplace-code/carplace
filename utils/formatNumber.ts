export default function formatNumber(value: number, options: { useThousandsSeparator?: boolean } = {}): string {
  if (typeof value !== "number") throw new TypeError("Value must be a number or Decimal instance");

  const { useThousandsSeparator } = options;
  const stringValue = value.toString();
  return useThousandsSeparator ? stringValue.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ".") : stringValue;
}
