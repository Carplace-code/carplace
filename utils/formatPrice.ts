export default function formatPrice(price: number | string): string {
  const localPrice = typeof price === "string" ? parseFloat(price.replace(/[^0-9.-]+/g, "")) : price;
  return Number(localPrice).toLocaleString("es-CL", {
    style: "currency",
    currency: "CLP",
  });
}
