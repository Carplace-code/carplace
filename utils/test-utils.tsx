// tests/test-utils.tsx
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render } from "@testing-library/react";
import { ReactNode } from "react";

// Puedes extender esta función si necesitas envolver en más contextos
export function renderWithQueryClient(children: ReactNode) {
  const queryClient = new QueryClient();
  return render(<QueryClientProvider client={queryClient}>{children}</QueryClientProvider>);
}
