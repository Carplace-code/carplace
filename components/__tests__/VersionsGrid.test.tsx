// components/__tests__/VersionsGrid.test.tsx
import { Decimal } from "@prisma/client/runtime/library";
import { describe, expect, it, vi } from "vitest";

import { renderWithQueryClient } from "@/utils/test-utils";

import type { MinimumGridVersion } from "../VersionCard";
import VersionsGrid from "../VersionsGrid";

// Mock de Clerk
vi.mock("@clerk/nextjs", () => ({
  useAuth: () => ({ isSignedIn: true }),
}));

describe("VersionsGrid", () => {
  it("renders a grid of version cards", () => {
    const mockVersions: MinimumGridVersion[] = [
      {
        id: "version-1",
        year: 2023,
        modelId: "model-1",
        model: {
          id: "model-1",
          name: "C200",
          bodyType: "sedan",
          brandId: "brand-1",
          brand: {
            id: "brand-1",
            name: "Mercedes-Benz",
          },
        },
        trims: [
          {
            id: "trim-1",
            name: "Base",
            motorSize: 2000,
            fuelType: "gas",
            transmissionType: "automatic",
            versionId: "version-1",
            carListings: [
              {
                id: "listing-1",
                sellerId: "seller-1",
                sourceId: "source-1",
                url: "https://example.com",
                title: "C200 2023",
                description: "En excelente estado",
                price: new Decimal(35990000),
                priceCurrency: "CLP",
                trimId: "trim-1",
                year: 2023,
                mileage: 10000,
                exteriorColor: "Silver",
                interiorColor: "Black",
                isNew: false,
                location: "Santiago",
                publishedAt: new Date(),
                scrapedAt: new Date(),
                images: [
                  {
                    id: "img-1",
                    url: "https://example.com/image.jpg",
                    listingId: "listing-1",
                  },
                ],
              },
            ],
          },
        ],
        wishlistItems: [],
      },
    ];

    const { container } = renderWithQueryClient(<VersionsGrid versions={mockVersions} />);
    expect(container).toMatchSnapshot();
  });

  it("renders message when no versions are available", () => {
    const { getByText } = renderWithQueryClient(<VersionsGrid versions={[]} />);
    expect(getByText("No hay versiones disponibles")).toBeInTheDocument();
  });
});
