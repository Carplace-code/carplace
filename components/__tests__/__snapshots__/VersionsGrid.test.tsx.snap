// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VersionsGrid > renders a grid of version cards 1`] = `
<div>
  <div
    class="grid w-full grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
  >
    <div
      class="rounded-lg border bg-white p-4 shadow-sm"
    >
      <div
        class="mb-4 h-32 overflow-hidden rounded-md bg-gray-200"
      >
        <img
          alt="C200"
          class="h-full w-full object-cover"
          data-nimg="1"
          decoding="async"
          height="128"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
      </div>
      <div>
        <h4
          class="text-xs font-semibold text-gray-500"
        >
          Mercedes-Benz
        </h4>
        <div
          class="flex w-full items-center justify-between"
        >
          <h3
            class="mb-1 font-semibold"
          >
            C200
          </h3>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-smfont-medium transition-all disabled:pointer-events-none disabled:opacity-50[&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0[&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px]aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 has-[>svg]:px-3 ml-2 h-auto p-1"
            data-slot="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-heart h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
              />
            </svg>
          </button>
        </div>
      </div>
      <p
        class="mb-2 text-sm text-gray-600"
      >
        2023
      </p>
      <p
        class="text-sm text-gray-500"
      >
        Gasolina
      </p>
      <a
        class="mt-4 block w-full rounded-md bg-blue-600 py-2 text-center text-sm text-white hover:bg-blue-700"
        href="/versions/version-1"
      >
        Ver detalles
      </a>
    </div>
  </div>
</div>
`;
